pkg/pipeline/extract/aggregate/aggregate.go

```diff
@@ -250,3 +255,66 @@ func (aggregate Aggregate) cleanupExpiredEntries() {
 		aggregate.GroupsList.Remove(listEntry)
 	}
 }
+
+// functions to manipulate a heap to generate TopK entries
+// We need to implement the heap interface: Len(), Less(), Swap(), Push(), Pop()
+
+type heapItem struct {
+	value   float64
+	metrics *config.GenericMap
+}
+
+type topkHeap []heapItem
+
+func (h topkHeap) Len() int {
+	return len(h)
+}
+
+func (h topkHeap) Less(i, j int) bool {
+	return h[i].value < h[j].value
+}
+
+func (h topkHeap) Swap(i, j int) {
+	h[i], h[j] = h[j], h[i]
+}
+
+func (h *topkHeap) Push(x interface{}) {
+	*h = append(*h, x.(heapItem))
+}
+
+func (h *topkHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	x := old[n-1]
+	*h = old[0 : n-1]
+	return x
+}
+
+func (aggregate Aggregate) computeTopK(inputMetrics []config.GenericMap) []config.GenericMap {
+	// maintain a heap with k items, always dropping the lowest
+	// we will be left with the TopK items
+	topk := aggregate.Definition.TopK
+	h := &topkHeap{}
+	for index, metricMap := range inputMetrics {
+		item := heapItem{
+			metrics: &inputMetrics[index],
+			value:   metricMap["total_value"].(float64),
+		}
+		heap.Push(h, item)
+		if h.Len() > topk {
+			heap.Pop(h)
+		}
```

Comment:
Not super critical but this could be optimized a bit further.
we "pay" `O(log (k))` for every Push and Pop operation. If we wrap this section with `if itemValue > minimum value of the heap` then we can save some cycles.